gitlab/resource_gitlab_project_freeze_period_test.go

```diff
@@ -0,0 +1,176 @@
+package gitlab
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabProjectFreezePeriod_basic(t *testing.T) {
+	var schedule gitlab.FreezePeriod
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectFreezePeriodDestroy,
+		Steps: []resource.TestStep{
+			// Create a project and freeze period with default options
+			{
+				Config: testAccGitlabProjectFreezePeriodConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectFreezePeriodExists("gitlab_project_freeze_period.schedule", &schedule),
+					testAccCheckGitlabProjectFreezePeriodAttributes(&schedule, &testAccGitlabProjectFreezePeriodExpectedAttributes{
+						FreezeStart:  "0 23 * * 5",
+						FreezeEnd:    "0 7 * * 1",
+						CronTimezone: "UTC",
+					}),
+				),
+			},
+			// Update the freeze period to change the parameters
+			{
+				Config: testAccGitlabProjectFreezePeriodUpdateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectFreezePeriodExists("gitlab_project_freeze_period.schedule", &schedule),
+					testAccCheckGitlabProjectFreezePeriodAttributes(&schedule, &testAccGitlabProjectFreezePeriodExpectedAttributes{
+						FreezeStart:  "0 23 * * 5",
+						FreezeEnd:    "0 7 * * 3",
+						CronTimezone: "UTC",
+					}),
+				),
+			},
+			// Update the freeze period to get back to initial settings
+			{
+				Config: testAccGitlabProjectFreezePeriodConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectFreezePeriodExists("gitlab_project_freeze_period.schedule", &schedule),
+					testAccCheckGitlabProjectFreezePeriodAttributes(&schedule, &testAccGitlabProjectFreezePeriodExpectedAttributes{
+						FreezeStart:  "0 23 * * 5",
+						FreezeEnd:    "0 7 * * 1",
+						CronTimezone: "UTC",
+					}),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabProjectFreezePeriodExists(n string, freezePeriod *gitlab.FreezePeriod) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		scheduleID := rs.Primary.ID
+		repoName := rs.Primary.Attributes["project_id"]
+		if repoName == "" {
+			return fmt.Errorf("No project ID is set")
+		}
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		freezePeriods, _, err := conn.FreezePeriods.ListFreezePeriods(repoName, nil)
```

Comment:
Is it possible to change this to use `GetFreezePeriod`
https://github.com/xanzy/go-gitlab/blob/master/freeze_periods.go#L81?

We wouldn't need to loop through an array if we were able to use the GetFreezePeriod function.

Thanks!